### PR TITLE
ENH: non-parametric test, similar to `ttest_1samp`

### LINKED ISSUE #17690: ENH: Feature Request - Addition of Non-parametric test based on Chebyshev's Inequality
### Is your feature request related to a problem? Please describe.

Whenever I need to perform a non-parametric test using Chebyshev's Inequality, I have to write the equation myself. It would be good to have it as a part of scipy.

### Describe the solution you'd like.

A function which takes in a distribution(array), a scalar value to be used for the test, the type of test(one-tailed/two-tailed), confidence and returns whether the scalar value is within the limits.

Chebyshev's inequality can also be used to define Confidence intervals which can be used to perform significance/hypothesis tests much like student's t-test. However, it is much more conservative than the student's t-test as **Chebyshev's inequality doesn't assume the distribution to be normal**.
Given below is the inequality,

&emsp;&emsp;&emsp;&emsp;&emsp;&emsp; $P(|X-µ| ≥ kσ) ≤ 1/k^2$

here, $P$ is the probability
&emsp;&emsp; $X$ is the random variable
&emsp;&emsp; $µ$ is the mean or expected value of the distribution
&emsp;&emsp; $σ$ is the standard deviation of the distribution
&emsp;&emsp; $k$ is an integer which can be used to find the probability of a value at $k$ standard deviations

A few papers where this inequality is used are mentioned below,
* [https://doi.org/10.1109/ICSPCC.2013.6663961](URL), Credibility test for blind processing results of sinusoid using Chebyshev's Inequality
* [https://doi.org/10.1002/for.3980080207](URL), An examination of the accuracy of judgemental confidence intervals in time series forecasting
* [https://doi.org/10.1111/j.1467-9876.2004.00428.x](URL), Chebyshev's inequality for nonparametric testing with small N and α in microarray research
* [https://doi.org/10.23919/FRUCT48808.2020.9087459](URL), Stream Data Preprocessing: Outlier Detection Based on the Chebyshev Inequality with Applications

There are many more papers where you can find this inequality being used especially when the distribution is not normal or if one wants to take a conservative approach to define the confidence intervals.

if anyone thinks this will be a useful feature, do let me know I can raise a pull request.

Cheers

### Describe alternatives you've considered.

Currently I have to write the code myself so I don't have any other alternative, would be better to have it as a part of scipy,

### Additional context (e.g. screenshots, GIFs)

_No response_

### PR DESCRIPTION
**Pull request by oshin94**
#### Reference issue

Closes #17690 

#### What does this implement/fix?

Non-parametric test to check if sample mean is same as a known population mean

#### Additional information

This is a draft PR. The tests for this piece of code have not been written yet.
@mdhaber @tupui @chrisb83 @tirthasheshpatel  Please see if I should add more content and the PR satisfies the checklists..
.
I have not created a separate class/object for the output as in `ttest_1samp`. Using the existing class `TtestResult` was not feasible as the method for computing the pvalues, statistics and confidence intervals are different and changes would have to be made in the methods. So, I avoided doing that for now. Though I think that would be possible, or a similar class can be defined. I am not sure what would have more utility. Please advice.

### CODE REVIEW

**Comment by mdhaber** [comment]
Thanks @oshin94. This is a good start. Go ahead and continue with this. One early suggestion is to refactor such that the returned `statistic` maintains the sign of the difference between means. You can take the `abs` if you need to when computing the confidence interval (which would be a separate method like it is for `ttest_1samp`).

**Comment by mdhaber** [comment]
@oshin94 no problem, but I'm just curious why you decided against continuing this one?

**Comment by mdhaber** [comment]
Sounds good. Please do answer https://github.com/scipy/scipy/issues/17690#issuecomment-1396527561 when you can, though.

**Comment by mdhaber** [comment]
There was no feedback from the [mailing list](https://mail.python.org/archives/list/scipy-dev@python.org/message/HXF6JZMMMWQATWREVCX3MJNYXDTB6NYQ/), and response in gh-17690 was lukewarm. We should probably look a bit further for support before continuing this one.

@raphaelvallat does this test look interesting to you (i.e. do you think it fills a gap)? Interesting enough that you'd want it in Pingouin?

@tupui does this look interesting enough to you that you'd be willing to review it?

File: scipy/stats/_stats_py.py [hunk 1]
```diff
@@ -10314,3 +10314,108 @@ def first_order(t):
     # finding a wrong root.
     res = root_scalar(first_order, x0=x0, x1=x1)
     return res.root
+
+
+def ChebyMeanTest_1samp(a, popmean, confidence_level=0.95,
+                        axis=0, nan_policy='propagate'):
```
Comment (tupui) [inline scipy/stats/_stats_py.py:10320]: We have a decorator for the `axis` and `nan_policy`: `_axis_nan_policy_factory`. I would either suggest using it, or ensure all associated tests would pass on this function.

Also optional parameters should be marked as keyword only (`*`).

Comment (tupui) [inline scipy/stats/_stats_py.py:10320]: `confidence_level` should be a method of the return object. See other tests such as `fligner` for e.g.

**Review by tupui — CHANGES_REQUESTED** [review]
Thank you for the PR @oshin94, it's a good start!

I left a few suggestions to get you started. These points would be necessary to make the function fit with SciPy's API.

The other very important thing which is missing will be tests. Here testing against reference values you could find in the literature would be preferred. Testing against other implementations you could find in R or elsewhere is also possible.

File: scipy/stats/_stats_py.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (oshin94) [reply scipy/stats/_stats_py.py:10320]: Yeah, `mean_test ` does makes sense.. will make the changes in the next PR..

[the remark this answers is not in the document]
Comment (oshin94) [reply scipy/stats/_stats_py.py:10320]: Yes, I did mean commit. sorry for the error.

**Comment by mdhaber** [comment]
@oshin94 just a reminder that you have a review waiting for you.

**Comment by mdhaber** [comment]
Hi @oshin94 thought I'd check in again. If I don't hear back in a week or so, I'll go ahead and close this, but you're always welcome to reopen.

**Pull request closed without merging**